src/exchanges/files.ts

```diff
@@ -0,0 +1,61 @@
+import { print } from 'graphql';
+import { extractFiles } from 'extract-files';
+import { filter, pipe, map } from 'wonka';
+import { Operation } from 'src/types';
+
+const fileExchange = ({ forward }) => {
+  const isMutation = (operation: Operation) =>
+    operation.operationName === 'mutation';
+
+  return ops$ => {
+    const preFlight$ = pipe(
+      ops$,
+      filter(isMutation),
+      map(operation => {
```

Comment:
Not quite sure about `map` here.

Reply:
I suppose you based this exchange on the example in issue #14 ? Over there, everything was done in the filter operation, working only with side effects instead of creating real operations.

I think it's better to base this on the official fetch-exchange, which is a bit more sophisticated in this regard.